lib/psych/visitors/to_ruby.rb

```diff
@@ -366,7 +366,7 @@ def revive_hash hash, o
               hash[key] = val
             end
           else
-            if @symbolize_names
+            if !force_string_keys && @symbolize_names
```

Comment:
`force_string_keys` is a misleading name. If you have a YAML doc that has Symbols, Numerics, Booleans, etc, as keys then setting `force_string_keys = true` passes those through without modification, it doesn't force them to be strings, it just skips the `#to_sym` call.

Reply:
Yes, I wasn't very insprired at that time, I think I'll rename it to `tagged` or something.